vector/src/main/java/im/vector/app/features/settings/VectorSettingsPreferencesFragment.kt

```diff
@@ -71,6 +76,17 @@ class VectorSettingsPreferencesFragment @Inject constructor(
             }
         }
 
+        findPreference<VectorSwitchPreference>(LightweightSettingsStorage.MATRIX_SDK_SETTINGS_PRESENCE_OFFLINE_MODE_ENABLED)!!.let { pref ->
+            pref.isChecked = lightweightSettingsStorage.getPresenceOfflineModeEnabled()
+            pref.setOnPreferenceChangeListener { _, newValue ->
+                val presenceOfflineModeEnabled = newValue as? Boolean ?: false
+                lifecycleScope.launch {
+                    session.setMyPresence(if (presenceOfflineModeEnabled) PresenceEnum.OFFLINE else PresenceEnum.ONLINE)
```

Comment:
Keep in mind there is also a 3rd value `PresenceEnum.UNAVAILABLE`, that I think we currently we are not using

Reply:
This call to `PresenceService` should be enough to tell the SDK to use `Offline` or `Online` for every sync request when on foreground